Modules/_ctypes/_ctypes.c

```diff
@@ -432,6 +433,24 @@ StructUnionType_paramfunc(CDataObject *self)
     parg->value.p = copied_self->b_ptr;
     parg->size = copied_self->b_size;
     parg->obj = (PyObject *)copied_self;
```

Comment:
I think this needs to be `PyTuple_Pack(2, self, copied_self)` for the case where we have both, along with correct refcounting. The rest of the change to this file should be safe to undo.

Reply:
@zooba, are you saying that the ref counting in the test code needs fixed (and the clang bindings code)?